fastlane/lanes/build.rb

```diff
@@ -315,6 +315,18 @@
     )
   end
 
+  lane :resolve_packages do |derived_data_path: DERIVED_DATA_PATH|
+    sh(
+      command: <<~CMD
+        xcodebuild \
+          -resolvePackageDependencies \
+          -workspace #{File.join(PROJECT_ROOT_FOLDER, 'WordPress.xcworkspace')} \
+          -scheme WordPress \
+          -derivedDataPath #{derived_data_path}
+      CMD
+    )
+  end
```

Comment:
I tried to use `build_app` with various `skip_*` flags, but couldn't get out of it running `xcodebuild build`.

I looked into the `xcodebuild` action, but it's a wrapper for canned commands that wouldn't have allowed to simply call this.

I made the derived data path a parameter with default because I find passing a value makes the localization code a bit clearer, see below.

Reply:
🎗️ Maybe worth creating an action for this in `release-toolkit` at some point (in a future, separate PR, of course)

That being said:
 - ⚠️  If I'm not mistaken, there's a risk for this invocation to update the package versions and the `Package.resolved` file (i.e. akin to `pod update` instead of `pod install`). You might want to also pass the `-onlyUsePackageVersionsFromResolvedFile` flag to this invocation to prevent this from happening.
 - ❓  I wonder if this lane and call is really needed in practice, given that the only lane that calls this is `generate_strings_file_for_glotpress`, and that lane is supposed to be run from CI, which already calls our `install_swiftpm_dependencies` helper from `ci-toolkit` before running the lane.
    - I guess it still makes sense to call it from the lane just as a way to ensure running a lane locally (e.g. for debugging purposes) would always result in the same outcome as running it on CI
    - OTOH, if the goal is only to _resolve_ SPM dependencies (as in, locally install the versions _pinned_ in the `Package.resolved`, as opposed to potentially _updating_ some of the Swift packages to a more recent version), I'd expect that to have already been done from opening the project in Xcode before testing the lane locally?

Reply:
Actually, I'd also recommend against using a string for passing the command, as this will make `sh` run the command through shell interpretation and interpolation, and thus make it subject to potential injection (accidental or not) and prone to issues with paths containing spaces.

It's safer to pass the command argument as separate arguments to `sh` instead to avoid those risks and pitfalls altogether:

```suggestion
  lane :resolve_packages do |derived_data_path: DERIVED_DATA_PATH|
    sh(
      'xcodebuild',
      '-resolvePackageDependencies', '-onlyUsePackageVersionsFromResolvedFile',
      '-workspace',  File.join(PROJECT_ROOT_FOLDER, 'WordPress.xcworkspace'),
      '-scheme', 'WordPress',
      '-derivedDataPath', derived_data_path
    )
  end
```

Reply:
@AliSoftware 

> OTOH, if the goal is only to resolve SPM dependencies (as in, locally install the versions pinned in the Package.resolved, as opposed to potentially updating some of the Swift packages to a more recent version), I'd expect that to have already been done from opening the project in Xcode before testing the lane locally?

Yes that's the goal and thank you for improving the implementation by suggesting the `-onlyUsePackageVersionsFromResolvedFile`.

The reason we need this automation is because the SPMs need to be checked out in CI, in particular in the lane that finalizes the code freeze. That lane does not build the app so we need to force fetching the sources of the SPM dependencies in the CI instance. So, while it's true that a dev should always have the sources in their machine, CI might not.

I made this a lane rather than a function or a call directly in the localization lane because it seemed handy to have it available, if just for testing.

Reply:
Got it. Makes more sense to me now 👍 

I wonder how we could ensure that the implementation of this lane doesn't risk being out of sync with the implementation of the same feature in `bin/install_swiftpm_dependencies` (e.g. I'm especially thinking about keeping the list of flags used when calling `xcodebuild` in sync) though, to avoid inconsistencies potentially popping up in the future around this implementation detail.

Maybe the long term solution (future PR) to ensure a single source of truth and avoid implementations from drifting away would be:
 - Make `install_swiftpm_dependencies` be able to take an extra parameter to specify the derived data path where to checkout the packages
 - Pass such parameter when calling `install_swiftpm_dependencies` in the `.buildkite/*.sh` scripts of this repo
 - Remove the `resolve_packages` lane, and replace it with a `UI.user_error!("Before running this translation lane locally, please ensure that you have asked Xcode to resolve SPM package dependencies first" if !is_ci && (code to check if package checkouts already exist in DerivedData path or not)`? 🤔